modal/functions.py

```diff
@@ -508,170 +734,146 @@ async def _preload(self: _Function, resolver: Resolver, existing_object_id: Opti
 
         async def _load(self: _Function, resolver: Resolver, existing_object_id: Optional[str]):
             assert resolver.client and resolver.client.stub
-            status_row = resolver.add_status_row()
-            status_row.message(f"Creating {tag}...")
-
-            if is_generator:
-                function_type = api_pb2.Function.FUNCTION_TYPE_GENERATOR
-            else:
-                function_type = api_pb2.Function.FUNCTION_TYPE_FUNCTION
+            with FunctionCreationStatus(resolver, tag) as function_creation_status:
```

Comment:
Ugh, this diff is mostly from reindentation I think... :( Let me see if I can extract the FunctionCreationStatus part into a separate PR to make this one easier to parse